test/parallel/test-runner-coverage-source-map.js

```diff
@@ -45,6 +45,31 @@ describe('Coverage with source maps', async () => {
     t.assert.strictEqual(spawned.code, 1);
   });
 
+  it('accounts only mapped lines when --enable-source-maps is provided', async (t) => {
+    const report = generateReport([
+      '# --------------------------------------------------------------',
+      '# file          | line % | branch % | funcs % | uncovered lines',
+      '# --------------------------------------------------------------',
+      '# a.test.ts     | 100.00 |   100.00 |  100.00 | ', // part of a bundle
+      '# b.test.ts     |  88.89 |   100.00 |  100.00 | 1', // part of a bundle
+      '# index.test.js |  71.43 |    66.67 |  100.00 | 6-7', // no source map
+      '# stdin.test.ts | 100.00 |   100.00 |  100.00 | ', // Source map without original file
```

Comment:
Maybe we should revert?

Reply:
I'm thinking that too

Reply:
I can work on it today guys so we have this quick fix. What do you say?

Reply:
If it is broken, we should revert immediately and re-land a fix once it's ready.

Reply:
The point is, if a line is not mapped it will be ignored anyways.

![image](https://github.com/user-attachments/assets/924510b2-a0a5-40c8-ad07-3afe268fabd4)

The function is not mapped by the source map, so it ignores. How should we proceed in this case? Would you have any clue @MoLow

Reply:
If source maps are enabled, coverage should be based on the original source. If a function in the original source is uncovered, it should be reported as uncovered.

I'm assuming the function is not mapped because of some dead code elimination or similar optimization during transpilation. That doesn't change the fact that it exists in the original source and is never executed.

Reply:
Makes sense. Thanks for the clarification.